Goutte/Client.php

```diff
@@ -121,7 +121,7 @@ protected function doRequest($request)
             $guzzleRequest->setHeader($name, $value);
         }
 
-        if ('POST' == $request->getMethod() || 'PUT' == $request->getMethod()) {
+        if (in_array($request->getMethod(), ['POST', 'PUT']) && $request->getFiles()) {
```

Comment:
You are right!
I've check 1000000 (million) times every variant:
```
== || == 0.24201488494873 seconds
in_array 0.77801990509033 seconds
```
However my variant is more readable